package htu21df;

message TempAndHumidity {
  optional int64 recorded_timestamp_ms = 1;
  optional double temp_degrees_c = 2;
  optional double percent_relative_humidity = 3;
  optional string sensor_name = 4;
  optional string debug = 5;
}

message DataAndClientId {
  optional string client_id = 1;
  optional TempAndHumidity temp_and_humidty_data = 2;
}

message UploadRequest {
  repeated DataAndClientId data_and_client_id = 1;
}

message UploadResponse {
  optional int32 num_saved = 1;
  repeated string unsaved_client_ids = 2;
}